assignment3/day 5/githubSearchAPI.yaml

```diff
@@ -0,0 +1,111 @@
+swagger: '2.0'
+info:
+  description: >-
+    The Search API is optimized to help you find the specific item you're
+    looking for (e.g., a specific user, a specific file in a repository, etc.).
+    Think of it the way you think of performing a search on Google. It's
+    designed to help you find the one result you're looking for (or maybe the
+    few results you're looking for). Just like searching on Google, you
+    sometimes want to see a few pages of search results so that you can find the
+    item that best meets your needs. To satisfy that need, the GitHub Search API
+    provides up to 1,000 results for each search.
+  version: 3.0.0
+  title: GitHub Search
+  contact:
+    email: vmishra@grepruby.com
+host: api.github.com
+basePath: /
+tags:
+  - name: Search
+    description: >-
+      Find repositories via various criteria. This method returns up to 100
+      results per page.
+    externalDocs:
+      description: Find out more
+      url: 'https://developer.github.com/v3/search/'
+schemes:
+  - https
+paths:
+  /search/repositories:
+    get:
+      tags:
+        - Search
+      summary: Find repositories via various criteria.
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
+        - name: q
+          in: query
+          description: 'The search keywords, as well as any qualifiers.'
+          required: false
+          type: string
+          items:
+            type: string
+            enum:
+              - author
+              - committer
+              - author-name
+              - committer-name
+              - author-email
+        - name: sort
+          in: query
+          description: >-
+            The sort field. One of stars, forks, or updated. Default: results
+            are sorted by best match.
+          required: false
+          type: string
+          items:
+            type: string
+            enum:
+              - starts
+              - forks
+              - updated
+        - name: order
+          in: query
+          description: 'The sort order if sort parameter is provided. One of asc or desc. '
+          required: false
+          type: string
+          items:
+            type: string
+            enum:
+              - asc
+              - dsc
+      responses:
+        '200':
+          description: OK
+          schema:
+            $ref: '#/definitions/SearchResult'
+definitions:
+  SearchResult:
+    type: object
+    properties:
+      total_count:
+        type: integer
+        description: It is a integer value.
+        example: 40
+      incomplete_results:
+        type: boolean
+        description: >-
+          It is a boolean type of value. true if incomplete results are possible
+          otherwise false.
+        example: false
+      items:
+        type: array
+        description: An array of properties.
+        example: >-
```

Comment:
_items_ is an array of object of a specific kind. Make another model in definitions for an element in _items_ array.